src/extractor/extraction_containers.cpp

```diff
@@ -567,41 +558,30 @@ void ExtractionContainers::WriteEdges(std::ofstream &file_out_stream) const
 
             // IMPORTANT: here, we're using slicing to only write the data from the base
             // class of NodeBasedEdgeWithOSM
-            NodeBasedEdge tmp = edge.result;
-            file_out_stream.write((char *)&tmp, sizeof(NodeBasedEdge));
-            used_edges_counter++;
+            normal_edges.push_back(edge.result);
```

Comment:
afaik we don't know how large the vector would be because of the conditional. Do you want to reserve for the size of `all_edges_list` and potentially overallocating?